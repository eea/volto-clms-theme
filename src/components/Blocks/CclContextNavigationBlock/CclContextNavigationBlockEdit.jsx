import React from 'react';
import ContextNavigationComponent from '@plone/volto/components/theme/Navigation/ContextNavigation';

const CclContextNavigationBlockEdit = props => {
    const regex = /\/edit$/;

  return <div>
            <strong>Navigation menu preview:</strong>
            <br /><br />
            <ContextNavigationComponent pathname={props.pathname.replace(regex, '')} />
        </div>;
};

export default CclContextNavigationBlockEdit;