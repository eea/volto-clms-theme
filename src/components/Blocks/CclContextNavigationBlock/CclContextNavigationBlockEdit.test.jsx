import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';
import CclContextNavigationBlockEdit from './CclContextNavigationBlockEdit';
import { MemoryRouter } from 'react-router-dom';
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

Enzyme.configure({ adapter: new Adapter() });

global.__SERVER__ = true; // eslint-disable-line no-underscore-dangle

const mockStore = configureStore();

describe('CclSearchBlock', () => {
  it('Check context navigation edit', () => {
    const store = mockStore({
      content: {
        create: {},
        data: {},
      },
      intl: {
        locale: 'en',
        messages: {},
      },
    });
    const pathname = '/example';
    const ContextNavigationBlockEdit = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <CclContextNavigationBlockEdit pathname={pathname}>
              <p>Search block view test</p>
            </CclContextNavigationBlockEdit>
          </MemoryRouter>
        </Provider>,
      )
      .toJSON();
    expect(ContextNavigationBlockEdit).toMatchSnapshot();
  });
});
