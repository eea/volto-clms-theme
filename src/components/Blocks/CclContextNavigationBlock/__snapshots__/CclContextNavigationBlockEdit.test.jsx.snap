// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CclSearchBlock Check context navigation edit 1`] = `
<div>
  <strong>
    Navigation menu preview:
  </strong>
  <br />
  <br />
  
</div>
`;
